src/feature/control/control.c

```diff
@@ -7206,9 +7212,7 @@ control_event_bootstrap_problem(const char *warn, const char *reason,
   if (we_are_hibernating())
     dowarn = 0;
 
-  while (status>=0 && bootstrap_status_to_string(status, &tag, &summary) < 0)
-    status--; /* find a recognized status string based on current progress */
-  status = bootstrap_percent; /* set status back to the actual number */
+  tor_assert(bootstrap_status_to_string(bootstrap_phase, &tag, &summary) == 0);
```

Comment:
After this removal, is `status` correct? It points to `bootstrap_percent` from the initial assignment and it is now never changed. Is this accurate since we seems to rely on the `bootstrap_phase` but still the status value is used in the log after ("Problem bootstrapping....") ?

Reply:
Yes, it should be correct. `status` now never gets modified by the loop, so it remains set to `bootstrap_percent` by its initializer. We use `bootstrap_phase` to get the milestone tag and summary, but we report `bootstrap_percent` (or equivalently, `status`) as the "percent" progress in the log message.

If it helps, I can eliminate the `status` variable and use `bootstrap_phase` and `bootstrap_percent` for clarity.